app/src/main/java/com/example/concordia_campus_guide/LocationFragment/LocationFragmentViewModel.java

```diff
@@ -91,49 +94,69 @@ private void setPolygonStyle(GeoJsonLayer layer, GoogleMap map){
 
             String[] coordinates = feature.getProperty("center").split(", ");
             LatLng centerPos = new LatLng(parseDouble(coordinates[1]), parseDouble(coordinates[0]));
-            addBuildingMarker(map, centerPos, Enum.valueOf(BuildingCode.class, feature.getProperty("code")));
+            addBuildingMarker(map, centerPos, feature.getProperty("code"), context);
         }
-        int x =213;
     }
 
     /**
      * The purpose of this method is to add a marker on the specified building.
      * @param map is the map used in our application.
      * @param centerPos is the latitude and longitude of the building's center
-     * @param buildingCode is the Building on which the method will add a marker
+     * @param buildingLabel is the Building on which the method will add a marker
      */
-    private void addBuildingMarker(GoogleMap map, LatLng centerPos, BuildingCode buildingCode) {
+    private void addBuildingMarker(GoogleMap map, LatLng centerPos, String buildingLabel, Context context) {
         Marker marker = map.addMarker(
                 new MarkerOptions()
                         .position(centerPos)
-                        .icon(BitmapDescriptorFactory.fromResource(getIcon(buildingCode)))
+                        .icon(setupMarker(buildingLabel,context))
                         .flat(true)
                         .anchor(0.5f,0.5f)
                         .alpha(0.90f)
                         //This line should be included whenever we test the UI for the marker:
                         //.title(buildingCode.toString())
         );
-        marker.setTag(buildingCode);
+        marker.setTag(buildingLabel);
     }
 
     /**
-     * Add classroom markers to map
-     * @param map
-     * @param centerPos
-     * @param classNumber
+     * The purpose of this method is setup the style of the marker.
+     * @param buildingLabel the label of the building
+     * @param context the context of the LocationFragment
+     * @return it will BitmapDescriptor object to use it as an icon for the marker on the map.
      */
-    private void addClassroomMarker(GoogleMap map, LatLng centerPos, String classNumber) {
-        Marker marker = map.addMarker(
-                new MarkerOptions()
-                        .position(centerPos)
-                        .icon(BitmapDescriptorFactory.fromResource(R.drawable.h))
-                        .flat(true)
-                        .anchor(0.5f,0.5f)
-                        .alpha(0.0f)
-        );
-        marker.setTag(classNumber);
+    private BitmapDescriptor setupMarker(String buildingLabel, Context context) {
+        int height = 150;
+        int width = 150;
```

Comment:
would be good to have them in our constant class or somewhere else since they seem to be a constant/set attribute